### PR TITLE
#1668 Fix subCriteria's not being updated

### PR DESCRIPTION
**Pull request by slyoldfox**
- [x] You have read the [Spring Data contribution guidelines](https://github.com/spring-projects/spring-data-build/blob/master/CONTRIBUTING.adoc).
- [x] There is a ticket in the bug tracker for the project in our [issue tracker](https://github.com/spring-projects/spring-data-elasticsearch/issues).
- [x] You use the code formatters provided [here](https://github.com/spring-projects/spring-data-build/tree/master/etc/ide) and have them applied to your changes. Don’t submit any formatting related changes.
- [x] You submit test cases (unit or integration tests) that back your changes.
- [x] You added yourself as author in the headers of the classes you touched. Amend the date range in the Apache license header if needed. For new types, add the license header (copy from another file and set the current year only).

### CODE REVIEW

**Comment by pivotal-issuemaster** [comment]
@slyoldfox Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-projects/spring-data-elasticsearch&pullRequestId=1670)!

[Click here](https://cla.pivotal.io/sync/spring?repositoryId=spring-projects/spring-data-elasticsearch&pullRequestId=1670) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/about) for frequently asked questions.

File: src/test/java/org/springframework/data/elasticsearch/core/CriteriaQueryMappingUnitTests.java [hunk 1]
```diff
@@ -109,6 +108,158 @@ void shouldMapNamesAndConvertValuesInCriteriaQuery() throws JSONException {
...
+	void shouldMapNamesAndConvertValuesInCriteriaQueryForSubCriteriaWithDate() throws JSONException {
+		Calendar calendar = Calendar.getInstance();
+		//calendar.set(1982, Calendar.FEBRUARY, 28, 13, 2, 1 );
+		calendar.setTimeInMillis(383745721653L);
+		// use POJO properties and types in the query building
+		CriteriaQuery criteriaQuery = new CriteriaQuery( //
+				Criteria.or()
+						.subCriteria( Criteria.where("birthDate") //
+								.between(LocalDate.of(1989, 11, 9), LocalDate.of(1990, 11, 9)) ) //
+						.subCriteria( Criteria.where( "createdDate").is(calendar.getTime()) ) //
```
Comment (sothawo) [inline src/test/java/org/springframework/data/elasticsearch/core/CriteriaQueryMappingUnitTests.java:197]:
```suggestion
						.subCriteria( Criteria.where( "createdDate").is(383745721653L) ) //
```
no need for the `Calendar` stuff here (I'd not use the old `java.util` date classes anyway, but always the classes from `java.time`).

File: src/test/java/org/springframework/data/elasticsearch/core/CriteriaQueryMappingUnitTests.java [hunk 2]
```diff
@@ -109,6 +108,158 @@ void shouldMapNamesAndConvertValuesInCriteriaQuery() throws JSONException {
 		assertEquals(expected, queryString, false);
 	}
 
+	@Test
+	// DATAES-1668
```
Comment (sothawo) [inline src/test/java/org/springframework/data/elasticsearch/core/CriteriaQueryMappingUnitTests.java:112]:
```suggestion
	// #1668
```
DATAES was when we were using Jira, now we just use the #NNN

File: src/test/java/org/springframework/data/elasticsearch/core/CriteriaQueryMappingUnitTests.java [hunk 3]
```diff
@@ -109,6 +108,158 @@ void shouldMapNamesAndConvertValuesInCriteriaQuery() throws JSONException {
...
+				"}"; //
+
+		mappingElasticsearchConverter.updateQuery(criteriaQuery, Person.class);
+		String queryString = new CriteriaQueryProcessor().createQuery(criteriaQuery.getCriteria()).toString();
+
+		assertEquals(expected, queryString, false);
+	}
+
+	@Test
+		// DATAES-1668
```
Comment (sothawo) [inline src/test/java/org/springframework/data/elasticsearch/core/CriteriaQueryMappingUnitTests.java:187]:
```suggestion
		// #1668
```

**Review by sothawo — CHANGES_REQUESTED** [review]
Thanks for this PR. It looks good, can you

* check and tick off the marks in the PR template (adding your name etc)
* sign the contributor's agreement
* rebase you branch on master an force push

the I can merge this

**Comment by pivotal-issuemaster** [comment]
@slyoldfox Thank you for signing the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-projects/spring-data-elasticsearch&pullRequestId=1670)!

**Comment by slyoldfox** [comment]
@sothawo Not a git wizard here - I hope I ticked off all the steps  as required :-)

**Pull request merged**